### PR TITLE
Check the response only has cacheable cookies before allowing caching

### PR DESCRIPTION
**Pull request by JordanMilne**
:eyeglasses: @ajacksified 

This makes sure people won't end up with the same CSRF token and such
if an intermediary proxy caches the response

### CODE REVIEW

File: src/server.jsx [hunk 1]
```diff
@@ -248,14 +253,33 @@ class Server {
...
       yield next;
     }
   }
 
+  cacheGuard (app) {
+    return function * (next) {
+
+      yield next;
+
+      var cookieHeaders = this.res.getHeader("Set-Cookie") || [];
```
Comment (ajacksified) [inline src/server.jsx]: single quotes

**Comment by ajacksified** [comment]
nit, and :fish:

**Comment by JordanMilne** [comment]
nit fixed^

**Comment by ajacksified** [comment]
https://github.com/reddit/reddit-mobile/commit/d7e7c10764e3eaa2812ef6071c328d1bc710f810

**Pull request closed without merging**